src/monty/io.py

```diff
@@ -54,7 +55,53 @@ def zopen(filename: Union[str, Path], *args, **kwargs) -> IO:
     return open(filename, *args, **kwargs)
 
 
-def reverse_readfile(filename: Union[str, Path]) -> Generator[str, str, None]:
+def _get_line_ending(
+    file: str | Path | io.TextIOWrapper,
+) -> Literal["\r\n", "\n", "\r"]:
+    """Helper function to get line ending of a file.
+
+    This function assumes the file has a single consistent line ending.
+
+    Returns:
+        "\n": Unix line ending.
+        "\r\n": Windows line ending.
+        "\r": Classic MacOS line ending.
+
+    Raises:
+        ValueError: If line ending is unknown.
+
+    Warnings:
+        If file is empty, "\n" would be used as default.
+    """
+    if isinstance(file, (str, Path)):
+        with open(file, "rb") as f:
+            first_line = f.readline()
```

Comment:
I think this makes the whole function very memory-expensive, I should rewrite it with `mmap` to read the last several chars instead the entire file.

Reply:
sounds like a good idea!